app.rb

```diff
@@ -0,0 +1,189 @@
+    @books << book
+    puts 'Book created successfully'
+  end
+
+  def list_all_books
+    @books.each_with_index { |book, index| puts "#{index}) Title: \"#{book.title}\", Author: \"#{book.author}\"" }
+  end
+
+  def list_all_people
+    @people.each_with_index do |person, index|
+      puts "#{index}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+  end
+
+  def read_desired_book
+    puts "\nSelect a book from the following list by number"
+    list_all_books
+    desired_book_index = gets.chomp
+    (0...@books.length).include?(desired_book_index.to_i) ? desired_book_index.to_i : read_desired_book
+  end
+
+  def read_desired_person
+    puts "\nSelect a person from the following list by number"
+    list_all_people
+    desired_person_index = gets.chomp
+    (0...@people.length).include?(desired_person_index.to_i) ? desired_person_index.to_i : read_desired_person
+  end
+
+  def read_desired_date
+    print "\nDate: "
+    gets.chomp
+  end
+
+  def create_rental
+    return print 'Please add a book first' if @books.empty?
+    return print 'Please add a person first' if @people.empty?
+
+    book = @books[read_desired_book]
+    person = @people[read_desired_person]
+    date = read_desired_date
+
+    rental = Rental.new(date, person, book)
+
+    @rentals << rental
+    puts 'Rental created successfully'
+  end
+
+  def create_for_user(user_input)
+    case user_input
+    when '3'
+      create_person
+    when '4'
+      create_book
+    when '5'
+      create_rental
+    end
+  end
+
+  def list_all_rentals_for_id
+    return puts 'Please add a rental first' if @rentals.empty?
+
+    puts "\nSelect a person from the following list by number"
+    @people.each do |person|
+      puts "ID: #{person.id}, [#{person.class}] Name: #{person.name}, Age: #{person.age}"
+    end
+    person_id = gets.chomp.to_i
+    rental_list = @rentals.select { |rental| rental.person.id == person_id }
+    if rental_list.empty?
+      puts 'No rentals found for this person.'
+    else
+      rental_list.each do |rental|
+        puts "Date: #{rental.date}, Book: \"#{rental.book.title}\" by  #{rental.book.author}"
+      end
+    end
+  end
+
+  def display_for_user(user_input)
+    case user_input
+    when '1'
+      list_all_books
+    when '2'
+      list_all_people
+    when '6'
+      list_all_rentals_for_id
+    end
+  end
+
+  def run
+    loop do
+      case user_input
+      when '1', '2', '6'
+        display_for_user(user_input)
+      when '3', '4', '5'
+        create_for_user(user_input)
+      when '7'
+        puts 'Thank you for using this app!'
+        exit(true)
+      else
+        puts "\nInvalid input \"#{user_input}\"!"
+        puts 'please try with one of these options:'
+        Utils.print_prompt
+        @user_input = gets.chomp
+        run
+      end
+      Utils.print_prompt
+      @user_input = gets.chomp
+    end
+  end
+end
+
+def main
+  app = App.new
+  app.run
+end
+
+main
```

Comment:
- [ ] Great job so far 👍 Your app functions perfectly and you are almost there. Kindly transfer this code to a `main.rb` file which will serve as the entry point to your console app when a user inputs `ruby main.rb`. You can read more on entry points [here](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/console_apps.md#console-app-entry-point).